runtime/src/main/java/com/dylibso/chicory/runtime/AotInterpreterMachine.java

```diff
@@ -0,0 +1,80 @@
+package com.dylibso.chicory.runtime;
+
+import com.dylibso.chicory.wasm.ChicoryException;
+import com.dylibso.chicory.wasm.types.FunctionType;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class AotInterpreterMachine extends InterpreterMachine {
+
+    private static final HashSet<Integer> usedInterpretedFunctions;
+
+    static {
+        if (Boolean.parseBoolean(
+                System.getProperty("chicory.compiler.printUseOfInterpretedFunctions"))) {
+            usedInterpretedFunctions = new HashSet<>();
```

Comment:
@andreaTP do you think we need to make this a synchronized set?

Reply:
No, not for now, functions are executed always in sequential order